ai_ta_backend/data_logging.py

```diff
@@ -0,0 +1,34 @@
+import os
+import nomic
+from nomic import atlas
+from langchain.embeddings import OpenAIEmbeddings
+import numpy as np
+import time
+
+class DataLog():
+    def __init__(self):
+        self.login = nomic.login(os.getenv('NOMIC_API_KEY'))
+
+    def nomic_log(self, course_name:str, search_query:str, retrieved_contexts)-> str:
+        """
+        Logs user query and retrieved contexts to Nomic.
+        """
+        try:
+            # convert query and context to embeddings
+            embeddings_model = OpenAIEmbeddings()
+            embeddings = embeddings_model.embed_query(search_query)

```

Comment:
Ok before I merge, please add print statements for the runtime of this whole nomic_log() function. 

Calls like this will double our latency numbers, at least, before we start returning answers to users, so we have to check before merge.

Reply:
Added this in the latest commit.

Reply:
need to actually run tests with postman and report results in seconds per nomic call.